cassandra-oss/files/entrypoint.sh

```diff
@@ -0,0 +1,201 @@
+        \! -user cassandra -exec chown cassandra '{}' +
+    # exec gosu cassandra "$BASH_SOURCE" "$@"
+fi
+
+_ip_address() {
+    # scrape the first non-localhost IP address of the container
+    # in Swarm Mode, we often get two IPs -- the container IP, and the (shared) VIP, and the container IP should always be first
+    ip address | awk '
+        $1 == "inet" && $NF != "lo" {
+            gsub(/\/.+$/, "", $2)
+            print $2
+            exit
+        }
+    '
+}
+
+# "sed -i", but without "mv" (which doesn't work on a bind-mounted file, for example)
+_sed-in-place() {
+    local filename="$1"; shift
+    local tempFile
+    tempFile="$(mktemp)"
+    sed "$@" "$filename" > "$tempFile"
+    cat "$tempFile" > "$filename"
+    rm "$tempFile"
+}
+
+_metrics_collector_supported() {
+    # currently, metrics collector does not work on arm64
+    [ "$(uname -m)" != "aarch64" ] && [ -z "$MGMT_API_DISABLE_MCAC" ]
+}
+
+if [ "$1" = 'mgmtapi' ]; then
+    echo "Starting Management API"
+
+    # Copy over any config files mounted at /config
+    # cp /config/cassandra.yaml /etc/cassandra/cassandra.yaml
+    if [ -d "/config" ] && ! [ "/config" -ef "$CASSANDRA_CONF" ]; then
+        cp -R /config/* "${CASSANDRA_CONF:-/etc/cassandra}"
+    fi
+
+    # Make sure the management api agent jar is set
+    # We do this here for the following reasons:
+    # 1. configbuilder will overwrite the cassandra-env-sh, so we don't want to set this after
+    # 2. We don't wan't operator or configbuilder to care so much about the version number or
+    #    the fact this jar even exists.
+    #
+    # MCAC metric filters are expected as an env variable in the following format:
+    # "deny:org.apache.cassandra.metrics.table allow:org.apache.cassandra.metrics.table.test"
+    
+    if _metrics_collector_supported && ! grep -qxF "JVM_OPTS=\"\$JVM_OPTS -javaagent:${MCAC_PATH}/lib/datastax-mcac-agent.jar\"" < ${CASSANDRA_CONF}/cassandra-env.sh ; then
+        # ensure newline at end of file
+        echo "" >> ${CASSANDRA_CONF}/cassandra-env.sh
+        echo "JVM_OPTS=\"\$JVM_OPTS -javaagent:${MCAC_PATH}/lib/datastax-mcac-agent.jar\"" >> ${CASSANDRA_CONF}/cassandra-env.sh
+        mkdir -p ${MCAC_PATH}
+        echo "" >> ${MCAC_PATH}/config/metric-collector.yaml
+        echo "data_dir_max_size_in_mb: 100" >> ${MCAC_PATH}/config/metric-collector.yaml
+        if [[ -n "$METRIC_FILTERS"  ]]; then
+            filter_array=(`echo $METRIC_FILTERS | sed 's/ /\n/g'`)
+            echo "filtering_rules:" >> ${MCAC_PATH}/config/metric-collector.yaml
+            for filter in "${filter_array[@]}"; do
+                echo "  - policy: $(echo ${filter} | cut -d':' -f1)" >> ${MCAC_PATH}/config/metric-collector.yaml
+                echo "    pattern: $(echo ${filter} | cut -d':' -f2)" >> ${MCAC_PATH}/config/metric-collector.yaml
+                echo "    scope: global" >> ${MCAC_PATH}/config/metric-collector.yaml
+            done
+        fi
+    fi
```

Comment:
I guess this needs to be removed or MCAC needs to be updated to work on Cassandra trunk